src/parse/Parser.ts

```diff
@@ -1,52 +1,82 @@
+import { BaseParseOpts } from 'types/ParseOptions';
 import { warnIfDebug } from 'utils/log';
-import { create, hasOwn } from 'utils/object';
 
 const leadingWhitespace = /^\s+/;
 
-const ParseError = function(message) {
-  this.name = 'ParseError';
-  this.message = message;
-  try {
-    throw new Error(message);
-  } catch (e) {
-    this.stack = e.stack;
+class ParseError extends Error {
+  public line: number;
+  public character: number;
+  public shortMessage: string;
+
+  constructor(message: string) {
+    super();
+    this.name = 'ParseError';
+    this.message = message;
   }
-};
+}
+
+export type Converter = (parser: Parser) => any;
+
+export interface CustomParser {
+  init: (str: string, options: BaseParseOpts) => void;
+  postProcess: (str: string, options?: BaseParseOpts) => any;
+}
+
+// todo add correct return types on props and methods
+// todo need to add parser type <T> and use it in result and postProcess function
+class Parser {
```

Comment:
Hi @evs-chris, 
I've rewritten `Parser` / `JsonParser` and `StandardParsers` using class syntax in order to have better code navigation during development of the various converters.
I've also added an interface that describe every item returned by converters function:
- `src/parse/converters/expressions/expressionDefinitions.ts`
- `src/parse/converters/templateItemDefinitions.ts`

Let me know if there is something that needs to be changed before I can move to other Parser sections.

Reply:
No, looks good!